tests/parametric/test_otel_span_methods.py

```diff
@@ -131,6 +132,73 @@ def test_otel_set_attributes_different_types(self, test_agent, test_library):
         assert root_span["metrics"]["d_int_val"] == 2
         assert root_span["metrics"]["d_double_val"] == 3.14
 
+    @irrelevant(context.library == "cpp", reason="library does not implement OpenTelemetry")
+    @missing_feature(context.library == "dotnet", reason="Array encoding not implemented")
+    @missing_feature(context.library < "java@1.22.0", reason="Array encoding not implemented")
+    @missing_feature(context.library == "golang", reason="Array encoding not implemented")
+    @missing_feature(context.library == "nodejs", reason="Empty string attribute value are not supported")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Array encoding not implemented")
+    @missing_feature(context.library == "python-http", reason="Array encoding not implemented")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    def test_otel_set_attributes_different_types_with_array_encoding(self, test_agent, test_library):
+        """
+            - Set attributes of multiple types for an otel span
+        """
+        start_time = int(time.time())
+        with test_library:
+            with test_library.otel_start_span("operation", span_kind=SK_PRODUCER, timestamp=start_time,) as span:
+                span.set_attributes({"str_val": "val"})
+                span.set_attributes({"str_val_empty": ""})
+                span.set_attributes({"bool_val": True})
+                span.set_attributes({"int_val": 1})
+                span.set_attributes({"int_val_zero": 0})
+                span.set_attributes({"double_val": 4.2})
+                span.set_attributes({"array_val_str": ["val1", "val2"]})
+                span.set_attributes({"array_val_int": [10, 20]})
+                span.set_attributes({"array_val_bool": [True, False]})
+                span.set_attributes({"array_val_double": [10.1, 20.2]})
+                span.set_attributes({"d_str_val": "bye", "d_bool_val": False, "d_int_val": 2, "d_double_val": 3.14})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, OtelSpan(name="operation"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        true_string = "True"
+        false_string = "False"
+        if root_span["meta"]["language"] in {"go", "jvm", "dotnet"}:
+            true_string = "true"
+            false_string = "false"
```

Comment:
Should we distinguish that? At the very least, OTel says that, the json representation of non-string values should be used. That would be just `true` and `false`.

Would also make sense to me to have that uniformly here, given that anything exported by OLTP will also be `true` and `false`, and not `True` and `False`.